### ISSUE TITLE
DJ decks and fx: corruption at the right of the screen in fmv under hardware mode when upscaling.

### ISSUE THREAD

**Issue by ghost** [PCSX2/pcsx2#2211]
***PCSX2 version:***
Last 1.5dev

***PCSX2 options:***
All default, speedhacks don't change the situation

***Plugins used:***
All default

***Description of the issue:***
The game "DJ decks and fx throw a corruption at the right of the screen under Hardware mode when upscaling, look:

![capture](https://user-images.githubusercontent.com/26351275/33807329-4029604e-ddd5-11e7-9943-ca5f35ad9525.PNG)

***Note:***
Native resolution work.

***How to reproduce the issue:***
Load DJ decks and fx, look a fmv under Hardware mode

***Not a regression***

***PC specifications:***
AMD fx8350 GTX 780Ti  Windows 10 64 Bit

***GSDUMP***
https://drive.google.com/file/d/1kZK4VNwAFyqSqtXrjt_sAGGQZrkekODD/view?usp=sharing

**Comment by MrCK1** [comment]
Did you try older versions of the emu? You're postive that it isn't a regression?

**Comment by atomic83GitHub** [comment]
Yes, I tried 1.4.0 and 1.2.1, still this issue unfortunatly.

**Comment by MrCK1** [comment]
Did you try all the SW/HW renderers as well?

**Comment by atomic83GitHub** [comment]
Yes, software don't have the issue but every hardware have this when using a non-native resolution (upscaling).

**Comment by MrCK1** [comment]
OK and is custom resolution affected as well?

**Comment by atomic83GitHub** [comment]
Not the same way, it just have a black border instead of the corruption:
![capture](https://user-images.githubusercontent.com/26351275/34029773-797633a2-e16a-11e7-9926-a277152460db.PNG)

**Comment by gregory38** [comment]
My guess is that something is wrong in the texture cache rescaling of old frame buffer (a wrong size maybe)

https://github.com/PCSX2/pcsx2/blob/master/plugins/GSdx/GSTextureCache.cpp#L1393
```
  m_renderer->m_dev->StretchRect(sTex, sRect, dTex, dRect, shader, linear);
```

**Comment by FlatOutPS2** [comment]
If this is the same issue I encountered in a few other games/demo's(I couldn't tell with the dump), then this is an old regression that occured sometime between PCSX2 v1.2.1 and a git build of v1.3.1 from january 2015. That regression is not related to the Texture Cache rescaling @gregory38 mentioned.

**Comment by RedDevilus** [comment]
Might be worth checking it out as gregory mentioned about old frame buffer and there has been a PR that takes in account the age of frames.

**Comment by refractionpcsx2** [comment]
Fixed with #6894 but also a duplicate of https://github.com/PCSX2/pcsx2/issues/4409
